org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/reflect/ConfigurationUtility.java

```diff
@@ -210,32 +217,44 @@ public static <T> T newInnerInstance(Object instance, Class<T> innerClass) {
    * @return true if the declared method is overwritten in implementationType
    */
   public static boolean isMethodOverwrite(Class<?> declaringType, String methodName, Class[] parameterTypes, Class<?> implementationType) {
-    Assertions.assertNotNull(declaringType, "declaringType must not be null");
-    Assertions.assertNotNull(methodName, "methodName must not be null");
-    Method declaredMethod;
-    try {
-      declaredMethod = declaringType.getDeclaredMethod(methodName, parameterTypes);
-    }
-    catch (NoSuchMethodException | SecurityException e) {
-      LOG.error("cannot find declared method {}.{}", declaringType.getName(), methodName, e);
+    assertNotNull(declaringType, "declaringType must not be null");
+    assertNotNull(methodName, "methodName must not be null");
+    Method declaredMethod = getDeclaredMethod(declaringType, methodName, parameterTypes);
+    if (declaredMethod == null) {
+      LOG.error("cannot find declared method {}.{}", declaringType.getName(), methodName);
       return false;
     }
     Class<?> c = implementationType;
     while (c != null && c != declaringType) {
-      try {
-        //check if method is avaliable
-        c.getDeclaredMethod(declaredMethod.getName(), declaredMethod.getParameterTypes());
+      Method method = getDeclaredMethod(c, declaredMethod.getName(), declaredMethod.getParameterTypes());
+      if (method != null) {
         return true;
       }
-      catch (NoSuchMethodException | SecurityException e) { // NOSONAR
-        //nop
-      }
-      //up
       c = c.getSuperclass();
     }
     return false;
   }
 
+  /**
+   * @return {@link Method} of class {@code clazz} with given {@code methodName} and {@code parameterTypes} or
+   * {@code null} if no matching method could be found.
+   */
+  public static Method getDeclaredMethod(Class<?> clazz, String methodName, Class[] parameterTypes) {
+    for (Method method : getAllDeclaredMethods(clazz)) {
+      if (method.getName().equals(methodName) && Arrays.equals(method.getParameterTypes(), parameterTypes)) {
+        return method;
+      }
+    }
+    return null;
+  }
+
```

Comment:
We could also introduce a cache for the method signature (declaring class, name, param types). Then the search loop would no longer be necessary?